### PR TITLE
Settings screen

### PR DESCRIPTION
**Pull request by DreadBoy**

### CODE REVIEW

**Comment by DreadBoy** [comment]
I'd like to discuss [WidgetLibrary](https://github.com/pebble-dev/mobile-app/pull/90/files#diff-4593dc3a70c365c43d57c65b3a85ccd3d57efa4e227f967cd84976a39a5fad5a). It's a screen with all custom components and is meant for developers to discover these custom components. Goal is to document everything so that new developers don't re-create existing components. I don't like having debugging screens in app but at the same time I think such WidgetLibrary is really useful. Can I get your opinions?

File: lib/ui/screens/alerting_apps.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+class _App {
+  final String name;
+  final bool enabled;
+
+  _App(this.name, this.enabled);
+}
+
+class AlertingApps extends HookWidget implements CobbleScreen {
+  final apps = [
```
Comment (matejdro) [inline lib/ui/screens/alerting_apps.dart:22]: :+1:

**Review by matejdro — APPROVED** [review]

**Comment by matejdro** [comment]
I like the idea of the widget library. Maybe button to the library could automatically show up on the bottom of the settings when debug version is opened?

Also, I would appreciate if someone that actually has any experience in flutter widgets would review this instead of me, I mostly checked out logic and found none (which is understandable since features behind this UI are not done yet).

File: lib/ui/common/components/cobble_sheet.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+///
+/// A modal bottom sheet prevents the user from interacting with the rest of
+/// the app until she dismisses it.
+///
+/// The [context] argument is used to look up the [Navigator] and [Theme] for
+/// the bottom sheet. It is only used when the method is called. Its
+/// corresponding widget can be safely removed from the tree before the bottom
+/// sheet is closed. Inside [builder] function you should use [context] provided
+/// to builder instead of one passed to [showCobbleSheet].
+Future<void> showCobbleSheet({
```
Comment (DreadBoy) [inline lib/ui/common/components/cobble_sheet.dart:14]: This component isn't implemented correctly. In fact there are 2 bottom sheet:

1. opens over bottom navigation, must be dismissed by clicking on background or dragging on handle
2. opens above bottom navigation (inside "tabview"), is persistent, doesn't have background and can't be dismissed. Also doesn't have a handle.

Current implementation support 1. use case only.

File: lib/ui/home/home_page.dart [hunk 3]
```diff
@@ -20,11 +21,11 @@ class _TabConfig {
 class HomePage extends HookWidget implements CobbleScreen {
   final _config = [
     _TabConfig(TestTab(), "Testing", RebbleIconsStroke.send_to_watch_checked),
-    _TabConfig(Placeholder(), "Health", RebbleIconsStroke.health),
+    _TabConfig(WidgetLibrary(), "Health", RebbleIconsStroke.health),
```
Comment (DreadBoy) [inline lib/ui/home/home_page.dart:24]: Don't do this, show extra entry on setting screen when app is in debug mode, instead.

File: lib/ui/common/components/cobble_sheet.dart (hunk 2 above)
Comment (DreadBoy) [reply lib/ui/common/components/cobble_sheet.dart]: Implemented 2. use case, too.

File: lib/ui/home/home_page.dart (hunk 3 above)
Comment (DreadBoy) [reply lib/ui/home/home_page.dart]: Implemented, now settings will have extra entry if app is in debug mode, called "Widget Library".

**Review by matejdro — APPROVED** [review]

**Pull request merged**